web/src/installers/index.ts

```diff
@@ -1143,6 +1160,14 @@ export class Installer {
     if (this.spec.kubernetes && this.spec.rke2) {
       return { error: { message: `This spec contains both kubeadm and rke2, please specifiy only one Kubernetes distribution` } };
     }
+    if (this.spec.k3s) {
+      if (!Installer.hasVersion("k3s", this.spec.k3s.version) && !this.hasS3Override("k3s")) {
+        return { error: { message: `K3S version ${_.escape(this.spec.k3s.version)} is not supported` } };
+      }
+    }
+    if (this.spec.kubernetes && this.spec.k3s) {
```

Comment:
well now dont you need to check all three?

Reply:
Yeah, I guess all possible combinations which means I need to probably do something different here. Going to merge this as is and think about the best way to do it.